apps/src/templates/PauseButton.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import color from '@cdo/apps/util/color';
+import {actions, selectors} from '@cdo/apps/lib/tools/jsdebugger/redux';
+import * as coreLibrary from '@cdo/apps/p5lab/spritelab/coreLibrary';
+
+const styles = {
+  button: {
+    minWidth: 0,
+    padding: '10px 13px',
+    backgroundColor: color.orange,
+    borderColor: color.orange,
+    color: color.white
+  }
+};
+
+class PauseButton extends React.Component {
+  static propTypes = {
+    // from redux
+    togglePause: PropTypes.func.isRequired,
+    isPaused: PropTypes.bool.isRequired,
+    isAttached: PropTypes.bool.isRequired
+  };
+
+  state = {
+    pauseStart: 0
+  };
+
+  togglePause = () => {
+    const current = new Date().getTime();
+    if (this.props.isPaused) {
+      coreLibrary.addPauseTime(current - this.state.pauseStart);
```

Comment:
this function should probably be a prop so that PauseButton doesn't depend on coreLibrary and can be reused